esp/services/ws_store/ws_storeService.cpp

```diff
@@ -385,22 +395,28 @@ bool CwsstoreEx::onFetchAll(IEspContext &context, IEspFetchAllRequest &req, IEsp
             storename = m_defaultStore.get();
     }
 
-    Owned<IPropertyTree> nstree = m_storeProvider->getAllPairs(storename, ns, secuser.get(), !req.getUserSpecific());
-
     IArrayOf<IEspKVPair> pairs;
-
-    Owned<IPropertyTreeIterator> iter = nstree->getElements("*");
-    ForEach(*iter)
     {
-        StringBuffer name;
-        StringBuffer value;
-        iter->query().getName(name);
-        nstree->getProp(name.str(), value);
-
-        Owned<IEspKVPair> kvpair = createKVPair("","");
-        kvpair->setKey(name.str());
-        kvpair->setValue(value.str());
-        pairs.append(*kvpair.getClear());
+        Owned<IPropertyTree> nstree = m_storeProvider->getAllPairs(storename, ns, secuser.get(), !req.getUserSpecific());
+
+        Owned<IPropertyTreeIterator> iter = nstree->getElements("*");
+        ForEach(*iter)
+        {
+            StringBuffer name;
+            StringBuffer value;
+            iter->query().getName(name);
+            nstree->getProp(name.str(), value);
+
+            Owned<IEspKVPair> kvpair = createKVPair("","");
+            
+            //it's possible this has been encoded, so decode it
+            StringBuffer decoded;
+            decodePtreeName(decoded, name.str());
+
+            kvpair->setKey(decoded.str());
```

Comment:
observation: No worth changing - but this breaks the encapsulation - the fact that the keys are encoded shouldn't really need to be known by the calling code.

Reply:
agreed, I do wish the encoding decoding could have been hidden from the client (ws_store) in this case.